internal/configs/configurator.go

```diff
@@ -148,10 +154,18 @@ func createUpstreamServerLabels(svcName string, resourceType string, resourceNam
 	return []string{svcName, resourceType, resourceName, resourceNamespace}
 }
 
+func createStreamUpstreamServerLabels(svcName string, resourceType string, resourceName string, resourceNamespace string) []string {
+	return []string{svcName, resourceType, resourceName, resourceNamespace}
+}
+
 func createServerZoneLabels(resourceType string, resourceName string, resourceNamespace string) []string {
 	return []string{resourceType, resourceName, resourceNamespace}
 }
 
+func createStreamServerZoneLabels(resourceType string, resourceName string, resourceNamespace string) []string {
+	return []string{resourceType, resourceName, resourceNamespace}
+}
```

Comment:
I agree these functions can be removed, all they do is return the parameters passed which is almost redundant because it can easily be done with one line of code and not calling a function e.g. ```labels = []string{label1, label2, label3......}```
I don't see how it would prevent bugs because there are no checks in these functions, they simply return the parameters ?

Reply:
These funcs serve the following purposes/bug preventions:
* Ensures the correct number of labelNames are set per nginx context. Prevents prometheus label cardinality error. (e.g: 3 label names, 4 label values)
* Ensures label names are always set in the correct order. Prevents label mismatch (e.g: resourceType=myResourceName)

From my perspective, I'm happy for this to be marked as resolved and proceed - we can take a look at a better solution at a later date. 

Possibly just removing these funcs and finding a way to add unit test coverage for future mistakes when extending labels.